pkg/repository/postgres/user.go

```diff
@@ -24,9 +24,9 @@ func NewUserRepository(shared *sharedRepository) repository.UserRepository {
 	}
 }
 
-func (w *userRepository) RegisterCreateCallback(callback repository.UnscopedCallback[*dbsqlc.User]) {
+func (w *userRepository) RegisterCreateCallback(callback repository.UserCreateCallback) {
 	if w.createCallbacks == nil {
-		w.createCallbacks = make([]repository.UnscopedCallback[*dbsqlc.User], 0)
+		w.createCallbacks = make([]repository.UserCreateCallback, 0)
```

Comment:
This is a breaking change. I know we need the OAuth information but just want to note it here. @abelanger5 I think it should be fine but we would need to update the cloud repo as well to adjust this change.